File: dotnet/Barchart.BinarySerializer/Schemas/Exceptions/InvalidEntityIdException.cs
namespace Barchart.BinarySerializer.Schemas.Exceptions;

/// <summary>
///     Thrown when an entity ID is invalid (i.e. [ 0 ]).
/// </summary>
public class InvalidEntityIdException : ArgumentException
{
    #region Constructor(s)
    
    /// <summary>
    ///     Initializes a new instance of the <see cref="InvalidEntityIdException"/> class.
    /// </summary>
    public InvalidEntityIdException() : base("Entity ID cannot be [ 0 ].")
    {
        
    }
    
    #endregion
}

File: typescript/src/schemas/exceptions/invalid-entity-id-exception.ts
/**
 * Thrown when an entity ID is invalid (i.e. [ 0 ]).
 *
 * @public
 * @exported
 * @extends {Error}
 */
export class InvalidEntityIdException extends Error {
    constructor() {
        super("Entity ID cannot be [ 0 ].");

        this.name = new.target.name;
    }
}
